python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -76,13 +80,30 @@ def __init__(
         # When the streaming generator ref is GC'ed, the objects it generated
         # cannot be reconstructed. Should remove it once Ray Core fixes the bug.
         self._finished_streaming_gens: List[StreamingObjectRefGenerator] = []
-        super().__init__(name, input_op)
+        super().__init__(name, input_op, target_max_block_size)
+
+        # If set, then all output blocks will be split into
+        # this many sub-blocks. This is to avoid having
+        # too-large blocks, which may reduce parallelism for
+        # the subsequent stage.
+        self._additional_split_factor = None
+
+    def get_additional_split_factor(self) -> int:
+        if self._additional_split_factor is None:
+            return 1
+        return self._additional_split_factor
+
+    def set_additional_split_factor(self, k: int):
```

Comment:
It seems that we can remove this "additional_split" concept now. Just changing the "target block size" should have the same effect?

Reply:
I tried to do this but unfortunately it didn't work because of #40246. Also, I think the additional split factor is still useful for cases where parallelism is explicitly set by the user. I'll add a TODO.